src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -256,10 +262,55 @@ void SensorFusion::updateWorld(const SSLProto::SSL_DetectionFrame &ssl_detection
         enemy_team    = createEnemyTeam(yellow_team);
     }
 
-    new_ball = createBall(ball_detections);
-    if (new_ball)
+    std::optional<Robot> robot_with_ball_in_dribbler;
+    if (friendly_robot_id_with_ball_in_dribbler.has_value())
+    {
+        robot_with_ball_in_dribbler =
+            friendly_team.getRobotById(friendly_robot_id_with_ball_in_dribbler.value());
+        friendly_robot_id_with_ball_in_dribbler = std::nullopt;
+    }
+    if (robot_with_ball_in_dribbler.has_value())
+    {
+        // use ball in dribbler information first since it's most precise
+        ball =
+            Ball(robot_with_ball_in_dribbler->position() +
+                     Vector::createFromAngle(robot_with_ball_in_dribbler->orientation())
+                         .normalize(DIST_TO_FRONT_OF_ROBOT_METERS),
+                 Vector(0, 0), robot_with_ball_in_dribbler->timestamp());
+    }
+    else
     {
-        updateBall(*new_ball);
+        std::optional<Ball> new_ball = createBall(ball_detections);
+        if (new_ball)
+        {
+            updateBall(*new_ball);
+        }
+        else if (ball)
+        {
+            // If we already have a ball, but miss the ball in the next frame
+            std::optional<Robot> closest_enemy =
+                enemy_team.getNearestRobot(ball->position());
+            std::optional<Robot> closest_friendly =
+                friendly_team.getNearestRobot(ball->position());
+
+            if (closest_friendly.has_value())
+            {
+                // it only makes sense to do anything if there are friendly robots
+                Robot closest_robot = closest_friendly.value();
+
+                if (closest_enemy.has_value() &&
+                    distanceSquared(closest_enemy->position(), ball->position()) <
+                        distanceSquared(closest_friendly->position(), ball->position()))
+                {
+                    closest_robot = closest_enemy.value();
+                }
+
+                ball = Ball(closest_robot.position() +
```

Comment:
looks like we're assuming the ball is always occluded by a dribbler. how reliable is this assumption?

Reply:
I was thinking about this too.

For example, I think checking the orientation of the robot with the last known ball position could be harmful (assuming enemy doesn't have possession when they could have it)

Reply:
the idea is that it'd be most threatening for robots to have the ball in dribbler, so we're assuming the worst case scenario